File: Step 07 Recursion/Step 7.1 Get a Strong Hold/Reverse a Stack/Reverse a Stack.cpp
//User function Template for C++

class Solution {
public:
    void insertAtBottom(stack<int> &st, int x) {
        if (st.empty()) st.push(x);
        else {
            int temp = st.top();
            st.pop();
            insertAtBottom(st, x);
            st.push(temp);
        }
    }
    void Reverse(stack<int> &st) {
        if (!st.empty()) {
            int x = st.top();
            st.pop();
            Reverse(st);
            insertAtBottom(st, x);
        }
    }
};

File: Step 07 Recursion/Step 7.1 Get a Strong Hold/Reverse a Stack/Reverse a Stack.java


//User function Template for Java

class Solution
{ 
    private static void insertAtBottom(Stack<Integer> s, int x) {
		if (s.isEmpty())
			s.push(x);
		else {
			int temp = s.pop();
			insertAtBottom(s, x);
			s.push(temp);
		}
	}
    static void reverse(Stack<Integer> s)
    {
        // add your code here
        if (!s.isEmpty()) {
            int temp = s.pop();
            reverse(s);
            insertAtBottom(s, temp);
        }
    }
}